torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,293 @@
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for biome, md5 in self.md5s.items():
+            download_url(
+                self.url.format(biome), self.root, md5=md5 if self.checksum else None
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        pathname = os.path.join(self.root, "*.tar.gz")
+        for tarfile in glob.iglob(pathname):
+            extract_archive(tarfile)
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image, mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        filepaths = cast(List[str], [hit.object for hit in hits])
+
+        if not filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        image_list: List[Tensor] = []
+        filename_regex = re.compile(self.filename_regex, re.VERBOSE)
+        for band in self.all_bands:
+            band_filepaths = []
+            for filepath in filepaths:
+                filename = os.path.basename(filepath)
+                directory = os.path.dirname(filepath)
+                match = re.match(filename_regex, filename)
+                if match:
+                    if "date" in match.groupdict():
+                        start = match.start("band")
+                        end = match.end("band")
+                        filename = filename[:start] + band + filename[end:]
+                        if band in ["B62", "B70", "B80"]:
+                            filename = filename.replace("L71", "L72")
+                filepath = os.path.join(directory, filename)
+                band_filepaths.append(filepath)
+            image_list.append(self._merge_files(band_filepaths, query))
+        image = torch.cat(image_list)
+
+        mask_filepaths = []
+        for filepath in filepaths:
+            path, row = os.path.basename(os.path.dirname(filepath)).split("_")[:2]
+            mask_filepath = filepath.replace(
+                os.path.basename(filepath), f"L7_{path}_{row}_newmask2015.TIF"
+            )
+            mask_filepaths.append(mask_filepath)
+
+        mask = self._merge_files(mask_filepaths, query)
+        mask_mapping = {64: 1, 128: 2, 191: 3, 255: 4}
```

Comment:
```suggestion
        mask_mapping = {64: 1, 128: 2, 192: 3, 255: 4}
```
We didn't catch this bug due to the aforementioned mistake in `data.py`

Reply:
The mask class is 191 instead of 192 in the 2015 version dataset.